js/controllers/wizard/fileUpload/form/FileUploadFormHandler.js

```diff
@@ -205,7 +205,10 @@
 
 		if (jsonData !== false) {
 			// Trigger the file uploaded event.
-			this.trigger('fileUploaded', jsonData.uploadedFile);
+			// If form validation fails, response does not have uploadedFile
```

Comment:
Isn't it enough to move the code below to the outer scope and conditionally call the `this.handleJson()` (only if `jsonData.status` is `true`)?

```js
if (jsonData.content !== '') {
   this.replaceWith(jsonData.content);
}
```

Reply:
handleJson is important also if status is false to display the error.

Reply:
I thought there was just this type of "HTML" errors on this form, but I see it has both 😁

Anyway, as you've pointed out, this pattern already exists, so no reason to come up with something new here.